### PR TITLE
Support resize on resource/request commodity

### PR DESCRIPTION
**Pull request by kevinwangcn**
# Intent
Improve resize test on request commodities in the integration test, also add memory case in.

# Background
The current integration test doesn't support resize test on request commodity

# Implementation
Compose resize action with memory adjustment and set the action type to assciate with request

# Test Done

## Case 1
**1. Build the binary and run the newly added case by executing this command**
```
make integration
./build/integration.test -k8s-kubeconfig=/root/.kube/config -k8s-context=kind-kind -ginkgo.focus="executing resize action on a deployment"
```
image

### CODE REVIEW

**Comment by enlinxu** [comment]
Is there a good reason why we introduce different method for request and limit with lots of duplicate code? If we introduce more commodities we may end up with more duplicate code. Let's please clean up the code to make it less duplicate.

**Comment by kevinwangcn** [comment]
> Is there a good reason why we introduce different method for request and limit with lots of duplicate code? If we introduce more commodities we may end up with more duplicate code. Let's please clean up the code to make it less duplicate.

I'm going to add a parameter in the function to differentiate the change type instead of adding a new function

**Review by enlinxu — APPROVED** [review]

File: test/integration/action_execution.go [hunk 1]
```diff
@@ -639,33 +678,88 @@ type mockProgressTrack struct{}
...
+		}
+		aiOnMem.CurrentSE = currentSE
 
-	newCap := cap + cpuIncrement
-	ai.NewComm = &proto.CommodityDTO{
-		CommodityType: &commType,
-		Capacity:      &newCap,
+		dto.ActionItem = []*proto.ActionItemDTO{aiOnCpu, aiOnMem}
+	default:
+		framework.Errorf("The change type<%d> doesn't get supported", changeType)
```
Comment (pankajbatra1) [inline test/integration/action_execution.go:760]: doesn't get -> is not

File: test/integration/action_execution.go [hunk 2]
```diff
@@ -192,20 +202,29 @@ var _ = Describe("Action Executor ", func() {
...
+			framework.ExpectNoError(err, "Resize action on limit failed")
 
-			_, err = waitForDeploymentToUpdateResource(kubeClient, dep)
+			_, err = waitForDeploymentToUpdateResource(kubeClient, dep, LIMIT_SINGLE_CONTAINER)
 			if err != nil {
-				framework.Failf("Failed to check the resource request in the new deployment: %s", err)
+				framework.Failf("Failed to check the change of resource/limit in the new deployment: %s", err)
+			}
+
+			// Resize on request, increasing cpu/memory request by 100m/100Mi
```
Comment (pankajbatra1) [inline test/integration/action_execution.go:221]: The comment here should not be specific to the actual test.

File: test/integration/action_execution.go [hunk 3]
```diff
@@ -639,33 +678,88 @@ type mockProgressTrack struct{}
...
-	commType := proto.CommodityDTO_VCPU
-	cap := float64(cpuIncrement)
-	ai.CurrentComm = &proto.CommodityDTO{
-		CommodityType: &commType,
-		Capacity:      &cap,
-	}
+	switch changeType {
+	case REQUEST_SINGLE_CONTAINER:
+		currentSE := newContainerEntity("test-cont")
+		// Build action item on Cpu
```
Comment (pankajbatra1) [inline test/integration/action_execution.go:687]: You could refactor the code to build an action item.  There are four almost identical code blocks.

File: test/integration/action_execution.go (hunk 1 above)
Comment (kevinwangcn) [reply test/integration/action_execution.go]: done

File: test/integration/action_execution.go (hunk 3 above)
Comment (kevinwangcn) [reply test/integration/action_execution.go]: Add a new function `newActionItemDTO` to build the action item,  please have a look at my second commit

File: test/integration/action_execution.go (hunk 2 above)
Comment (kevinwangcn) [reply test/integration/action_execution.go]: done

**Review by irfanurrehman — APPROVED** [review]

**Review by pankajbatra1 — APPROVED** [review]

**Pull request merged**